clang/lib/Sema/SemaType.cpp

```diff
@@ -2730,15 +2726,8 @@ QualType Sema::BuildVectorType(QualType CurType, Expr *SizeExpr,
   }
 
   if (!TypeSize || VectorSizeBits % TypeSize) {
-    // Disable errors for SYCL and OpenMP device since definition of __bf16 is
-    // being moved to a shared header and it causes new errors emitted when
-    // host code is compiled with device compiler for SPIR target.
-    // FIXME: device code specific diagnostic is probably needed.
-    if (!(!TypeSize &&
-          (getLangOpts().OpenMPIsDevice || getLangOpts().SYCLIsDevice))) {
       Diag(AttrLoc, diag::err_attribute_invalid_size)
           << SizeExpr->getSourceRange();
-    }
     return QualType();
   }
```

Comment:
Does this match with community now?

Reply:
Yes, please see https://github.com/intel/llvm/pull/7186/commits/79509df3e233a522a72932070e1e2fccec53041e .